internal/pkg/service/buffer/api/service/service.go

```diff
@@ -186,6 +191,45 @@ func (*service) Import(dependencies.ForPublicRequest, *buffer.ImportPayload, io.
 	return &NotImplementedError{}
 }
 
+func parseRequestBody(contentType string, reader io.ReadCloser) (res *orderedmap.OrderedMap, err error) {
+	// Limit read csv to 1 MB plus 1 B. If the reader fills the limit then the request is bigger than allowed.
+	limitedReader := io.LimitReader(reader, configstore.MaxImportRequestSizeInBytes+1)
+	defer func() {
+		if closeErr := reader.Close(); closeErr != nil && err == nil {
+			err = errors.Errorf("cannot close request body reading: %w", closeErr)
+		}
+	}()
+
+	buf := new(strings.Builder)
+	_, err = io.Copy(buf, limitedReader)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check that the reader did not read more than the maximum.
+	if buf.Len() > configstore.MaxImportRequestSizeInBytes {
+		return nil, &PayloadTooLargeError{Message: "Payload too large."}
+	}
```

Comment:
Nevšiml jsem si, že by bylo něco jako LimitedReader, který by vrátil chybu pokud do něj nateče víc dat, tak jsem to obešel tím že povolím načíst o 1B víc než máme limit a pokud se ten limit o ten 1B překročí, tak vyhodíme chybu.